src/components/GameItem.js

```diff
@@ -4,26 +4,45 @@ import ItemIcons from '../ItemIcons.js';
 import PropTypes from 'prop-types';
 
 class GameItem extends Component {
+  constructor() {
+    super();
 
+    this.state = {
+      gameItemClass: "game-item",
+    };
+  }
 
   render() {
+
     const itemStyle = {
       bottom: `${this.props.height}px`, // use props.height to offset from the bottom of screen
       zIndex: this.props.layer, // use props.layer to set z-index, so we display ontop of background
     };
 
+    const markAsSpotted = () => {
+      if (this.props.type === 'litter') {
+        this.setState({
+          gameItemClass: this.state.gameItemClass + ' spotted-litter'
+        });
+        this.props.onItemClicked();
+      } else {
+        this.setState({
+          gameItemClass: this.state.gameItemClass + ' spotted-nature'
+        });
+      }
+    }
+
     // Update this to select the correct icon for each item
-    const icon = ItemIcons.rock;
+    const icon = ItemIcons[`${this.props.type}`];
 
     return (
-      <div className="game-item" style={itemStyle}>
+      <div className={ this.state.gameItemClass } style={itemStyle} onClick={ markAsSpotted }>
         <img src={icon} alt="Item" className="icon-item"></img>
       </div>
     );
   }
 }
 
-
```

Comment:
I suggest adding your callback and type to `propTypes`.

Reply:
✅